configure

```diff
@@ -583,7 +583,7 @@ def host_arch_cc():
     # would be xlc so hard code gcc
     k = cc_macros('gcc')
   else:
-    k = cc_macros()
+    k = cc_macros(os.environ.get('CC_host', None))
```

Comment:
You can drop the second argument, `None` is the default value when the key is not found.  LGTM apart from that.